app/src/sharedTest/java/org/oppia/app/profile/PinPasswordActivityTest.kt

```diff
@@ -709,6 +705,31 @@ class PinPasswordActivityTest {
     }
   }
 
+  // TODO(#1840)
+  /**
+   * Appends the specified text to a view. This is needed because Robolectric doesn't seem to
+   * properly input digits for text views using 'android:digits'. See
+   * https://github.com/robolectric/robolectric/issues/5110 for specifics.
+   */
+  private fun appendText(text: String): ViewAction {
+    val typeTextViewAction = typeText(text)
+    return object : ViewAction {
+      override fun getDescription(): String = typeTextViewAction.description
+
+      override fun getConstraints(): Matcher<View> = typeTextViewAction.constraints
+
+      override fun perform(uiController: UiController?, view: View?) {
+        // Appending text only works on Robolectric, whereas Espresso needs to use typeText().
+        if (Build.FINGERPRINT.contains("robolectric", ignoreCase = true)) {
+          (view as? EditText)?.append(text)
```

Comment:
this indentation looks incorrect to me?

Reply:
no, I guess. I had already done the reformat thing.

Reply:
Okay thanks